Source/Core/VideoBackends/OGL/TextureConverter.cpp

```diff
@@ -242,34 +242,24 @@ static void EncodeToRamUsingShader(GLuint srcTexture, u8* destAddr, u32 dst_line
 
   int dstSize = dst_line_size * dstHeight;
 
-  if ((writeStride != dst_line_size) && (dstHeight > 1))
+  // When the dst_line_size and writeStride are the same, we could use glReadPixels directly to RAM.
+  // But instead we always copy the data via a PBO, because macOS inexplicably prefers this for some
+  // reason.
+  glBindBuffer(GL_PIXEL_PACK_BUFFER, s_PBO);
+  glBufferData(GL_PIXEL_PACK_BUFFER, dstSize, nullptr, GL_STREAM_READ);
```

Comment:
It's a safe assumption to allocate this buffer upfront as 1024 * 1024 * 4 (max size of a texture in guest memory), and then re-use it, rather than reallocating it each time. Only a small optimization, though I doubt it'd make any measurable difference since you're forcing the GPU to idle by the readback anyway. The Vulkan backend (and possibly D3D, I would have to check) do it this way.